docs/addons/jwt.md

```diff
@@ -0,0 +1,312 @@
+# JWT Authentication
+
+> **Note**
+> Shield now supports only JWS (Singed JWT). JWE (Encrypted JWT) is not supported.
+
+## What is JWT?
+
+JWT or JSON Web Token is a compact and self-contained way of securely transmitting
+information between parties as a JSON object. It is commonly used for authentication
+and authorization purposes in web applications.
+
+For example, when a user logs in to a web application, the server generates a JWT
+token and sends it to the client. The client then includes this token in the header
+of subsequent requests to the server. The server verifies the authenticity of the
+token and grants access to protected resources accordingly.
+
+If you are not familiar with JWT, we recommend that you check out
+[Introduction to JSON Web Tokens](https://jwt.io/introduction) before continuing.
+
+## Setup
+
+To use JWT Authentication, you need additional setup and configuration.
+
+### Manual Setup
+
+1. Install "firebase/php-jwt" via Composer.
+
+    ```console
+    composer require firebase/php-jwt:^6.4
+    ```
+
+2. Copy the **AuthJWT.php** from **vendor/codeigniter4/shield/src/Config/** into your project's config folder and update the namespace to `Config`. You will also need to have these classes extend the original classes. See the example below.
+
+    ```php
+    // new file - app/Config/AuthJWT.php
+    <?php
+
+    declare(strict_types=1);
+
+    namespace Config;
+
+    use CodeIgniter\Shield\Config\AuthJWT as ShieldAuthJWT;
+
+    /**
+     * JWT Authenticator Configuration
+     */
+    class AuthJWT extends ShieldAuthJWT
+    {
+        // ...
+    }
+    ```
+
+3. If your **app/Config/Auth.php** is not up-to-date, you also need to update it. Check **vendor/codeigniter4/shield/src/Config/Auth.php** and apply the differences.
+
+## Configuration
+
+Configure **app/Config/AuthJWT.php** for your needs.
+
+### Set the Default Claims
+
+> **Note**
+> A payload contains the actual data being transmitted, such as user ID, role,
+> or expiration time. Items in a payload is called *claims*.
+
+Set the default payload items to the property `$defaultClaims`.
+
+E.g.:
+```php
+    public array $defaultClaims = [
+        'iss' => 'https://codeigniter.com/',
+    ];
+```
+
+The default claims will be included in all tokens issued by Shield.
+
+### Set Secret Key
+
+Set your secret key in the `$keys` property, or set it in your `.env` file.
+
+E.g.:
+```dotenv
+authjwt.keys.default.0.secret = 8XBFsF6HThIa7OV/bSynahEch+WbKrGcuiJVYPiwqPE=
+```
+
+It needs more than 256 bits random string. You can get a secure random string
+with the following command:
```

Comment:
I think we should show this in a larger context so that users have no doubt about how it works.

```suggestion
It needs at least 256 bits random string. The length of the secret depends on the 
algorithm we use. The default one is `HS256`, so to ensure that the hash value is 
secure and not easily guessable, the secret key should be at least as long as the 
hash function's output - 256 bits (32 bytes). You can get a secure random string 
with the following command:
```

Reply:
Thanks! It is true.